### DISCUSSION TITLE
Why doesn't the Textcat have a tok2vec?

### DISCUSSION THREAD

**Discussion by taghouti-ghofrane** [explosion/spaCy#9702] [Help: Model Advice] [answered]
Hello all , i am confused in generating base_config.cfg from spacy documentation!
when i select components textcat , hardware:cpu , optimize for : efficiency ,
the pipeline has just Textcat: 

[nlp]
lang = "en"
pipeline = ["textcat"]
batch_size = 1000

Is it Normal ?

How we can Train a model without Tok2vec in The pipeline ?
I trained the model just with that Pipeline , it didnt give good result !

* Python Version Used: 3.9.0
* spaCy Version Used: 3.2

**Comment by adrianeboyd** [comment, marked as answer]
The simplest textcat model uses a bag of words only, so it doesn't require a `tok2vec`, but as you noticed, the performance is not always as good as some of the other (less efficient) options. In that case, you can try other architectures instead, like `spacy.TextCatEnsemble.v2`:

https://spacy.io/api/architectures#textcat
